src/User/Password/PasswordAlgorithm.php

```diff
@@ -0,0 +1,63 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MsgPhp\User\Password;
+
+/**
+ * @author Roland Franssen <franssen.roland@gmail.com>
+ */
+final class PasswordAlgorithm
+{
+    private const DEFAULT_LEGACY = 'sha512';
+
+    /** @var int|string */
+    public $type;
+
+    /** @var array|null */
+    public $options;
+
+    /** @var string|null */
+    public $salt;
+
+    /** @var bool */
+    public $legacy;
+
+    public static function create(int $type = \PASSWORD_DEFAULT, array $options = []): self
+    {
+        $instance = new self();
+        $instance->type = $type;
+        $instance->options = $options;
+        $instance->legacy = false;
+
+        return $instance;
+    }
+
+    public static function createLegacy(string $type = self::DEFAULT_LEGACY): self
+    {
+        $instance = new self();
+        $instance->type = $type;
+        $instance->legacy = true;
+
+        return $instance;
+    }
+
+    public static function createSaltedLegacy(string $salt, string $type = self::DEFAULT_LEGACY): self
+    {
+        $instance = new self();
+        $instance->type = $type;
+        $instance->salt = $salt;
+        $instance->legacy = true;
+
+        return $instance;
+    }
+
+    public function __toString(): string
+    {
+        return (string) $this->type;
+    }
+
+    private function __construct()
```

Comment:
Seems like `$type` and `$legacy` are mandatory parameters, could be moved to the constructor to reflect this

Reply:
Now less LOC!